### PR TITLE
fix: paginate dashboards

### PR DESCRIPTION
**Pull request by drdelambre**
prevents large organizations from being crippled after they import their 1.x dashboards using pkgr

the technique is to add dashboard cards one at a time until the user's window is filled, use that number to represent page size (without connecting css rules + js code), and then to render a few pages.

when the user gets to the bottom of the last page, add another one, until there are no more pages to add. it has the downside of getting slow as the user scrolls to the bottom, but we can always nudge user behavior towards using the filter mechanism (keeping it always visible, type ahead, whatevs) if the need presents itself.

this PR also removes some prop drilling that was going on in the area, sorry for the clutter.

### CODE REVIEW

File: ui/src/dashboards/components/DashboardsCardGrid.scss [hunk 1]
```diff
@@ -3,12 +3,25 @@
...
+  }
+}
+
 .dashboards-card-grid {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 150px 150px 150px;
    grid-column-gap: $ix-marg-a;
    grid-row-gap: $ix-marg-a;
+   height: 100%;
```
Comment (Asalem1) [inline ui/src/dashboards/components/DashboardsCardGrid.scss]: You’re 100%

File: ui/src/dashboards/components/dashboard_index/DashboardCards.tsx [hunk 2]
```diff
@@ -28,15 +26,110 @@ export default class DashboardCards extends PureComponent<Props> {
     getSortedResources
   )
 
+  state = {
+    hasMeasured: false,
+    pages: 1,
+    windowSize: 0,
+  }
+
+  public componentDidMount() {
```
Comment (Asalem1) [inline ui/src/dashboards/components/dashboard_index/DashboardCards.tsx]: If the state is being set initially, why are we using componentDidMount to set it again?

Comment (drdelambre) [reply ui/src/dashboards/components/dashboard_index/DashboardCards.tsx]: it's for when the component gets unmounted (like if you type in a filter that has no matches), and then remounted (remove the filter), to restart the measuring cycle

File: ui/src/dashboards/components/DashboardsCardGrid.scss (hunk 1 above)
Comment (drdelambre) [reply ui/src/dashboards/components/DashboardsCardGrid.scss]: it was the worst drilling 100% height all the way to the frame component, or else it just chooses whatever height it wants to be

File: ui/webpack.dev.ts [hunk 3]
```diff
@@ -11,6 +11,10 @@ module.exports = merge(common, {
   output: {
     filename: '[name].js',
   },
+  watchOptions: {
```
Comment (drdelambre) [inline ui/webpack.dev.ts]: this was to make webpack work in docker on linux

File: ui/src/dashboards/components/dashboard_index/DashboardCards.tsx (hunk 2 above)
Comment (Asalem1) [reply ui/src/dashboards/components/dashboard_index/DashboardCards.tsx]: Very interesting. I'll allow it

**Pull request merged**